### PR TITLE
Fix stubtest issues

### PR DESCRIPTION
**Pull request by hatal175**

### CODE REVIEW

File: tests/stubtest_whitelists/linux.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ grp.getgrgid
 grp.struct_group._asdict
 grp.struct_group._make
 grp.struct_group._replace
+mmap.MADV_SOFT_OFFLINE  # Not defined in some linuxes
```
Comment (Akuli) [inline tests/stubtest_whitelists/linux.txt:13]: Is this needed for CI or for running stubtest on your computer? I think `.github/workflows/stubtest-unused-whitelist.yml` will delete this automatically if the CI doesn't need it.

**Comment by hatal175** [comment]
Failed on CI.
See
https://github.com/hatal175/typeshed/runs/2520806762?check_suite_focus=true

File: tests/stubtest_whitelists/linux.txt (hunk 1 above)
Comment (Akuli) [reply tests/stubtest_whitelists/linux.txt:13]: Also, `MADV_SOFT_OFFLINE` is new on Linux 2.6.33 (see `madvise(2)` manual page), so unless something hasn't been updated for the past 15 years, this shouldn't be a problem.

**Comment by Akuli** [comment]
This turned out to be a bug in glibc. I compiled my own Python 3.9 and edited `Modules/mmapmodule.c` like this:

```diff
@@ -1623,6 +1624,8 @@ PyInit_mmap(void)
 #endif
 #ifdef MADV_SOFT_OFFLINE
     setint(dict, "MADV_SOFT_OFFLINE", MADV_SOFT_OFFLINE);
+#else
+   #error its missing lol
 #endif
 #ifdef MADV_HUGEPAGE
     setint(dict, "MADV_HUGEPAGE", MADV_HUGEPAGE);
```

Then it didn't compile anymore. The problem is that it's not in any of the includes that `#include <sys/mman.h>` brings along, unlike other MADV constants which are mostly in `bits/mman-linux.h`.

Nobody has reported this bug yet. I was going to report this by clicking "File a bug" [here](https://www.gnu.org/software/libc/involved.html), but they they want an email address and they have a warning about receiving spam to it.

I think we should delete `MADV_SOFT_OFFLINE` from the stub and add a comment mentioning how broken it is.

File: tests/stubtest_whitelists/py38.txt [hunk 2]
```diff
@@ -34,7 +34,6 @@ collections.abc.AsyncGenerator.ag_running
 collections.abc.ItemsView.__reversed__
 collections.abc.KeysView.__reversed__
 collections.abc.ValuesView.__reversed__
-concurrent.futures.ProcessPoolExecutor.map
```
Comment (Akuli) [inline tests/stubtest_whitelists/py38.txt:37]: I wonder why this wasn't deleted last Saturday when `stubtest-unused-whitelist.yml` ran

**Review by Akuli — APPROVED** [review]

**Pull request merged**